articles/client-auth/v2/mobile-desktop.md

```diff
@@ -0,0 +1,444 @@
+* `code_challenge`: the generated challenge associated with the `code_verifier`.
+* `code_challenge_method`: the method used to generate the challenge. The PKCE spec defines two methods (`S256` and `plain`), but Auth0 only supports `S256`.
+
+::: panel-info Arbitrary Claims
+To improve Client application compatibility, Auth0 returns profile information using an [OIDC-defined structured claim format](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims). This means that arbitrary claims to ID or access tokens must conform to a namespaced format to avoid collisions with standard OIDC claims. For example, if your namespace is `https://foo.com/` and you want to add an arbitrary claim named `myclaim`, you would name the claim `https://foo.com/myclaim`, not `myclaim`.
+:::
+
+As an example, your HTML snippet for your authorization URL might look as follows:
+
+```html
+<a href="https://${account.namespace}/authorize?
+  audience=appointments:api&
+  scope=appointments%20contacts&
+  response_type=code&
+  client_id=${account.clientId}&
+  code_challenge=E9Melhoa2OwvFrEMTJguCHaoeK1t8URWbuGJSstw-cM&
+  code_challenge_method=S256&
+  redirect_uri=com.myclientapp://myclientapp.com/callback">
+  Sign In
+</a>
+```
+
+If all goes well, you'll receive an HTTP 302 response:
+
+```text
+HTTP/1.1 302 Found
+Location: https://YOUR_APP/callback?code=AUTHORIZATION_CODE
+```
+
+Note the authorization code included at the end of the included URL.
+
+### Obtain an ID Token
+
+Using the authorization code obtained in step 2, you can obtain the ID token by making the appropriate `POST` call to the [tokens endpoint](api/authentication#authorization-code-pkce-).
+
+```har
+{
+  "method": "POST",
+  "url": "https://${account.namespace}/oauth/token",
+  "headers": [
+    { "name": "Content-Type", "value": "application/json" }
+  ],
+  "postData": {
+    "mimeType": "application/json",
+    "text": "{\"grant_type\":\"authorization_code\",\"client_id\": \"${account.clientId}\",\"code_verifier\": \"YOUR_GENERATED_CODE_VERIFIER\",\"code\": \"YOUR_AUTHORIZATION_CODE\",\"redirect_uri\": \"com.myclientapp://myclientapp.com/callback\", }"
+  }
+}
+```
+
+Request Parameters:
+
+* `grant_type`: Set this field to `authorization_code`
+* `client_id`: Your application's Client ID
+* `code_verifier`: The cryptographically random key used to generate the `code_challenge` passed to the `/authorize` endpoint
+* `code`: The authorization code received from the initial `authorize` call
+* `redirect_uri`: the `redirect_uri` passed to `/authorize` (these two values must match exactly)
+
+If all goes well, you'll receive an HTTP 200 response with the following payload:
+
+```json
+{
+  "access_token":"eyJz93a...k4laUWw",
+  "refresh_token":"GEbRxBN...edjnXbL",
+  "id_token":"eyJ0XAi...4faeEoQ",
+  "token_type":"Bearer",
+  "expires_in":86400
+}
+```
+
+Note that Auth0 includes the `refresh_token` only if you've:
```

Comment:
I would remove references to `Enable **Allow Offline Access** for your API`. We didn't tell them anywhere in this doc to create an API, this is only for user authentication, not API authorization.

I would add a note though that the `access_token` can be used to call the `/userinfo` endpoint of the Authentication API https://auth0.com/docs/api/authentication#get-user-info